### PR TITLE
Replace endpoint.ParseHostPort with net.SplitHostPort to fix IPv6 client endpoints

### PR DESCRIPTION
**Pull request by jpbetz**
Fixes https://github.com/kubernetes/kubernetes/issues/83550

Manually verified fix:

```
etcd --listen-client-urls="https://[::1]:2379" --advertise-client-urls="https://[::1]:2379" --cert-file=integration/fixtures/server-ipv6.crt --key-file=integration/fixtures/server-ipv6.key.insecure

etcdctl --cacert ~/projects/etcd-io/src/go.etcd.io/etcd/integration/fixtures/server-ipv6.crt --endpoints="https://[::1]:2379" get / --write-out json
{"header":{"cluster_id":14841639068965178418,"member_id":10276657743932975437,"revision":1,"raft_term":5}}
```

We need a automated test for this too. Since we don't have any IPv6 tests in etcd, we need to enable that test carefully, since it would fail in environments without IPv6 enabled. We will do that as a follow up PR.

Regenerated all the test certs after adding one with an IPv6 IP for the manual test. We'll keep the cert around for when we add the automated test.

cc @liggitt @gyuho @jingyih @wenjiaswe @YoyinZyc @hangxie

### CODE REVIEW

**Comment by neolit123** [comment]
LGTM, thanks for the quick fix.

**Comment by neolit123** [comment]
given kubernetes/kubernetes is at 3.3.15 would it be possible to also backport for 3.3?

**Comment by wenjiaswe** [comment]
lgtm, thanks Joe! +1 on backporting to 3.3, ref: https://github.com/kubernetes/kubernetes/pull/83209#issuecomment-539673564

File: clientv3/client.go [hunk 1]
```diff
@@ -372,12 +375,15 @@ func (c *Client) directDialCreds(ep string) grpccredentials.TransportCredentials
 			// Set the server name must to the endpoint hostname without port since grpc
 			// otherwise attempts to check if x509 cert is valid for the full endpoint
 			// including the scheme and port, which fails.
-			host, _ := endpoint.ParseHostPort(hostPort)
+			host, _, err := net.SplitHostPort(hostPort)
+			if err != nil {
```
Comment (liggitt) [inline clientv3/client.go]: this returns err if the hostPort had no port... is that what we want? ParseHostPort tolerated that previously

Comment (jpbetz) [reply clientv3/client.go]: Thanks. Fixed.

Comment (liggitt) [reply clientv3/client.go]: lgtm. Also, looks like the return signature doesn't need to change now.

**Pull request merged**